source/Player.hx

```diff
@@ -187,6 +210,48 @@ import flixel.group.FlxGroup;
     btmBox.y = y + height - hitBoxHeight;
   }
 
+  //Override player.kill to drop any weilded items
+  override public function kill():Void 
+  {
+	  if (weilding){
+		dropCurrentEquip();
+	  }
+	  super.kill();
+	  new FlxTimer().start(2, resetLevel, 1); //This helps speed things up for debugging
+  }
+  
+  private function resetLevel(Timer:FlxTimer):Void
+	{
+		FlxG.resetState();
+	}
+
+	/**
+	 * Method to pickup items. If the player is not holding anything
+	 * and the item is weildable, equip the item. Otherwise add it to his bag.
+	 * If player is not weilding anything and the item is weildable
+	 *	set weilding to true and equip
+	*/
+	public function pickup_item(player:Player, item:Item):Void {
+		if (!weilding && item.weildable){
+			if(!item.justDropped){
+				weilding = true;
+				equipped_item = item;
+				item.equip(this);
+			}
+		} 
+	}
+
+	/**
+	 * Method to drop the currently equipped item
+	 * 
+	 */
+	private function dropCurrentEquip(){
+		equipped_item.drop_item();
+		weilding = false;
+		equipped_item = null;
+	}
+}
```

Comment:
This bracket needs removed.  It currently closes the class before getting to `bounce()`